libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -112,18 +130,16 @@ bool DeclarationTypeChecker::visit(StructDefinition const& _struct)
 		for (ASTPointer<VariableDeclaration> const& member: _struct.members())
 		{
 			Type const* memberType = member->annotation().type;
-			while (auto arrayType = dynamic_cast<ArrayType const*>(memberType))
-			{
-				if (arrayType->isDynamicallySized())
-					break;
-				memberType = arrayType->baseType();
-			}
+
```

Comment:
Does this change disallow recursion via dynamic arrays even in storage? (it should still be allowed)

Reply:
No. The change should be equivalent. The Boolean parameter for `finalBaseType()` will decide if we continue after encountering a dynamic array.